org.adempiere.base.process/src/org/compiere/process/OrderCreateProduction.java

```diff
@@ -0,0 +1,164 @@
+
+package org.compiere.process;
+
+import java.util.List;
+import java.util.logging.Level;
+
+import org.compiere.model.MDocType;
+import org.compiere.model.MOrder;
+import org.compiere.model.MOrderLine;
+import org.compiere.model.MProduct;
+import org.compiere.model.MProduction;
+import org.compiere.model.MWarehouse;
+import org.compiere.model.Query;
+import org.compiere.util.Env;
+import org.compiere.util.Msg;
+
+/**
+ *
+ * @author Carlos Ruiz - globalqss - bxservice
+ *
+ */
+@org.adempiere.base.annotation.Process
+public class OrderCreateProduction extends SvrProcess {
+	/** Order */
+	private int p_C_Order_ID = 0;
+
+	/**
+	 * Prepare - e.g., get Parameters.
+	 */
+	protected void prepare() {
+		p_C_Order_ID = getRecord_ID();
+	} // prepare
+
+	/**
+	 * Create Production for BOM products of an order
+	 *
+	 * @throws Exception
+	 */
+	protected String doIt() throws Exception {
+		if (log.isLoggable(Level.INFO))
+			log.info("C_Order_ID=" + p_C_Order_ID);
+		if (p_C_Order_ID == 0)
+			throw new IllegalArgumentException("No Order");
+		//
+		MOrder order = new MOrder(getCtx(), p_C_Order_ID, get_TrxName());
+		if (order.get_ID() == 0)
+			throw new IllegalArgumentException("Order not found");
+		if (!MOrder.DOCSTATUS_Completed.equals(order.getDocStatus()))
+			throw new IllegalArgumentException("Order not completed");
+
+		final String where = "C_OrderLine.C_Order_ID=?"
+				+ " AND C_OrderLine.Processed='Y'"
+				+ " AND p.IsBOM='Y'"
+				+ " AND p.IsVerified='Y'"
+				+ " AND NOT EXISTS (SELECT 1 FROM M_Production pr WHERE pr.C_OrderLine_ID=C_OrderLine.C_OrderLine_ID)";
+		List<MOrderLine> lines = new Query(getCtx(), MOrderLine.Table_Name, where, get_TrxName())
+				.addJoinClause("JOIN M_Product p ON (C_OrderLine.M_Product_ID=p.M_Product_ID)")
+				.setOnlyActiveRecords(true)
+				.setParameters(p_C_Order_ID)
+				.list();
+
+		MDocType doc = new MDocType(getCtx(), order.getC_DocType_ID(), get_TrxName());
+		int cnt = 0;
+		for (MOrderLine line : lines) {
+			if ((line.getQtyOrdered().subtract(line.getQtyDelivered())).compareTo(Env.ZERO) <= 0) {
+				if (!doc.getDocSubTypeSO().equals("ON")) { // Consignment and stock orders both have subtype of ON
+					continue;
+				}
+			}
+
+			MProduction production = new MProduction(line);
+			MProduct product = new MProduct(getCtx(), line.getM_Product_ID(), get_TrxName());
+
+			production.setM_Product_ID(line.getM_Product_ID());
+			production.setProductionQty(line.getQtyOrdered().subtract(line.getQtyDelivered()));
+			production.setDatePromised(line.getDatePromised());
+			if (product.getM_Locator_ID() > 0)
+				production.setM_Locator_ID(product.getM_Locator_ID());
+			production.setC_OrderLine_ID(line.getC_OrderLine_ID());
+
+			int locator = product.getM_Locator_ID();
+			if (locator == 0)
+				locator = MWarehouse.get(getCtx(), line.getM_Warehouse_ID()).getDefaultLocator().get_ID();
+			production.setM_Locator_ID(locator);
```

Comment:
this make line 101 & 102 above redundant

Reply:
Thanks @hengsin - I copied the code from OrderLineCreateProduction and didn't notice the redundant assignment there.

So, with the last commit I refactored also that class and additionally removed the unused parameters